warehouse/age-off-utils/src/main/java/datawave/age/off/util/AgeOffCsvToMatchPatternFormatter.java

```diff
@@ -0,0 +1,212 @@
+
+        appendValue(sb, tokens);
+
+        sb.append(NEW_LINE);
+
+        this.writer.write(sb.toString());
+    }
+
+    private void appendValue(StringBuilder sb, String[] tokens) {
+        String value = "";
+
+        // use override value if it exists for this line (it might be empty)
+        if (configuration.useOverrides()) {
+            if (tokens.length <= columnInformation.overrideColumnNumber) {
+                log.error("Unable to process override {}", Arrays.toString(tokens));
+                throw new IllegalStateException("Not enough tokens");
+            }
+            value = tokens[columnInformation.overrideColumnNumber].trim();
+        }
+
+        // if overrides are disabled or override was missing
+        if (value.length() == 0) {
+            if (tokens.length <= columnInformation.durationColumnNumber) {
+                log.error("Unable to process duration {}", Arrays.toString(tokens));
+                throw new IllegalStateException("Not enough tokens");
+            }
+            value = tokens[columnInformation.durationColumnNumber].trim();
+        }
+
+        if (value.length() == 0) {
+            log.error("Unable to find non-empty override or duration {}", Arrays.toString(tokens));
+            throw new IllegalStateException("Empty token");
+        }
+        sb.append(attemptValueMapping(value));
+    }
+
+    private String attemptValueMapping(String originalValue) {
+        if (null == configuration.getValueMapping()) {
+            return originalValue;
+        }
+
+        String replacementValue = configuration.getValueMapping().get(originalValue);
+        if (null == replacementValue) {
+            return originalValue;
+        }
+        return replacementValue;
+    }
+
+    private void appendLabel(String[] tokens, StringBuilder sb) {
+        if (configuration.shouldDisableLabel()) {
+            return;
+        }
+
+        if (tokens.length <= columnInformation.labelColumnNumber) {
+            log.error("Unable to process label {}", Arrays.toString(tokens));
+            throw new IllegalStateException("Not enough tokens");
+        }
+
+        String label = "";
+
+        if (null != configuration.getStaticLabel()) {
+            label = configuration.getStaticLabel();
+        } else if (columnInformation.labelColumnNumber != -1) {
+            label = tokens[columnInformation.labelColumnNumber].trim();
+        }
+
+        if (label.length() == 0) {
+            log.error("Unable to apply non-empty label {}", Arrays.toString(tokens));
+            throw new IllegalStateException("Empty token");
+        }
+        sb.append(label).append(SPACE);
+    }
+
+    private void appendLiteral(String[] tokens, StringBuilder sb) {
+        if (tokens.length <= columnInformation.patternColumnNumber) {
+            log.error("Unable to process literal {}", Arrays.toString(tokens));
+            throw new IllegalStateException("Not enough tokens");
```

Comment:
Instead of logging and throwing, what do you think about adding this context to the IllegalStateException. I think logging part of the error and throwing the Exception could potentially lead to the error message and the stacktrace being separated. Like if the caller catches the IllegalStateException and chooses not to log it for instance.

It could look like:
```
throw new IllegalStateException("Not enough tokens when parsing literal from {}" ...);
```

or something like that.

Reply:
Addressed in latest push.